libcontainer/nsenter/nsexec.c

```diff
@@ -163,8 +163,10 @@ static void write_log(const char *level, const char *format, ...)
 		json = NULL;
 		goto out;
 	}
-
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-result"
 	write(logfd, json, ret);
+#pragma GCC diagnostic pop
```

Comment:
This looks way too heavy to silent a single warning.

I think something like #3168 looks more elegant; PTAL.